JabbR/Commands/GiphyCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using JabbR.Models;
+using Microsoft.Ajax.Utilities;
+using Microsoft.AspNet.SignalR;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using RestSharp;
+using Http = JabbR.Infrastructure.Http;
+
+namespace JabbR.Commands
+{
+    [Command("giphy","Giphy_CommandInfo", "keyword1 keyword2 keywordX", "user")]
+    public class GiphyCommand : UserCommand
+    {
+        public override void Execute(CommandContext context, CallerContext callerContext, ChatUser callingUser, string[] args)
+        {
+            if (String.IsNullOrEmpty(callerContext.RoomName))
+            {
+                throw new HubException(LanguageResources.InvokeFromRoomRequired);
+            }
+
+            ChatRoom callingRoom = context.Repository.GetRoomByName(callerContext.RoomName);
+
+            string url = String.Format("http://api.giphy.com/v1/gifs/random?api_key={0}&tag=", "dc6zaTOxFJmzC");
```

Comment:
Should the API key be injected into this ctor or something? so then it can exist in the `<appSettings>` ... which are in azure so the data isn't public. ??